src/board.rs

```diff
@@ -84,18 +88,18 @@ impl Board {
     }
 
     pub fn move_piece(&mut self, m: &Move) -> ChessResult<()> {
-        self.is_move_valid(&m)?;
-        let mut piece = self.get_piece(&m.from);
-
-        if let Some(Piece::Pawn(player)) = piece {
-            if (player == Player::White && m.to.y == 7) || (player == Player::Black && m.to.y == 0)
-            {
-                // TODO: always promote to queen for now, need to handle this eventually
-                piece = Some(Piece::Queen(player));
+        self.is_move_valid(m)?;
+
+        if let Some(mut piece) = self.get_piece_mut(&m.from).take() {
```

Comment:
So if we know it is a valid move at this point, would it be cleaner to `let piece ... .unwrap()`?

Reply:
No because I want to borrow the piece mutably and then change it

Reply:
Note the `mut piece `

Reply:
Yeah I mean `let mut piece ...`

Reply:
`let mut piece = self.get_piece_mut(&m.from).take().unwrap();`